### PR TITLE
String extensions

### PR DESCRIPTION
**Pull request by n0an**
Added new methods and var to String Extensions:
- public var isSpelledCorrectly: Bool
- func deletingPrefix()
- func deletingSuffix()

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [ ] I have added a changelog entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/SwiftStdlib/StringExtensions.swift [hunk 1]
```diff
@@ -856,7 +865,28 @@ public extension String {
...
 	}
-	
+    
+    /// SwifterSwift: Deletes given prefix from string.
+    ///
+    ///   "Hello, World!".deletingPrefix("Hello, ") -> "World!"
+    ///
+    /// - Parameter prefix: Prefix to delete from string.
+    /// - Returns: The string after prefix deleting.
+    public func deletingPrefix(_ prefix: String) -> String {
```
Comment (SD10) [inline Sources/Extensions/SwiftStdlib/StringExtensions.swift]: I like these extensions 👏 Do you think we should name it `removingPrefix`?

Comment (n0an) [reply Sources/Extensions/SwiftStdlib/StringExtensions.swift]: Yes, maybe "removing" is better word than "deleting". I checked there's "remove" word used more than "delete" in code base. Should I change methods names to removingPrefix and removingSuffix ?

Comment (SD10) [reply Sources/Extensions/SwiftStdlib/StringExtensions.swift]: @n0an Sounds good to me 👌 Please add CHANGELOG entries as well

**Comment by SD10** [comment]
@n0an You added the CHANGELOG entries to the wrong section. Can you update them?

**Comment by n0an** [comment]
Yes, I will fix, but which section is correct?
Should I add my new entries to "Upcoming release/Added" section?

**Comment by SD10** [comment]
Just used these in my project 💯

**Pull request merged**